examples/petsc/src/petscutils.c

```diff
@@ -79,47 +79,54 @@ PetscErrorCode Kershaw(DM dm_orig, PetscScalar eps) {
 // Create BC label
 // -----------------------------------------------------------------------------
 static PetscErrorCode CreateBCLabel(DM dm, const char name[]) {
-  int ierr;
+
   DMLabel label;
 
   PetscFunctionBeginUser;
 
-  ierr = DMCreateLabel(dm, name); CHKERRQ(ierr);
-  ierr = DMGetLabel(dm, name, &label); CHKERRQ(ierr);
-  ierr = DMPlexMarkBoundaryFaces(dm, 1, label); CHKERRQ(ierr);
+  PetscCall(DMCreateLabel(dm, name));
+  PetscCall(DMGetLabel(dm, name, &label));
+  PetscCall(DMPlexMarkBoundaryFaces(dm, PETSC_DETERMINE, label));
+  PetscCall(DMPlexLabelComplete(dm, label));
 
   PetscFunctionReturn(0);
 };
 
 // -----------------------------------------------------------------------------
 // This function sets up a DM for a given degree
 // -----------------------------------------------------------------------------
-PetscErrorCode SetupDMByDegree(DM dm, PetscInt degree, PetscInt num_comp_u,
+PetscErrorCode SetupDMByDegree(DM dm, PetscInt p_degree, PetscInt q_extra,
+                               PetscInt num_comp_u,
                                PetscInt dim, bool enforce_bc, BCFunction bc_func) {
   PetscInt ierr, marker_ids[1] = {1};
+  PetscInt q_degree = p_degree + q_extra;
   PetscFE fe;
   MPI_Comm comm;
+  PetscBool      is_simplex = PETSC_TRUE;
 
   PetscFunctionBeginUser;
 
+  // Check if simplex or tensor-product mesh
+  ierr = DMPlexIsSimplex(dm, &is_simplex); CHKERRQ(ierr);
   // Setup FE
   ierr = PetscObjectGetComm((PetscObject)dm, &comm); CHKERRQ(ierr);
-  ierr = PetscFECreateLagrange(comm, dim, num_comp_u, PETSC_FALSE, degree, degree,
-                               &fe); CHKERRQ(ierr);
+  ierr = PetscFECreateLagrange(comm, dim, num_comp_u, is_simplex, p_degree,
```

Comment:
In Ratel we started calling this 'order' over 'degree'

Reply:
If we want to make it like ratel, I think it is better to change it in a follow-up PR, since all the examples in libCEED are using `degree`. Maybe we need to change the name of some function  like SetupDMByDegree,... or we can keep libCEED as is.